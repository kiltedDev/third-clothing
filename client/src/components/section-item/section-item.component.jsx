import React from 'react';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cart.actions';

import {
  SectionItemContainer,
  SectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer
} from './section-item.styles';

const SectionItem = ({ item, addItem }) => {
  const { name, price, imageUrl } = item;

  return (
    <SectionItemContainer className='section-item'>
      <BackgroundImage className='image' imageUrl={imageUrl} />
      <SectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </SectionFooterContainer>
      <AddButton onClick={() => addItem(item)} inverted>
        Add to cart
      </AddButton>
    </SectionItemContainer>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(
  null,
  mapDispatchToProps
)(SectionItem);
