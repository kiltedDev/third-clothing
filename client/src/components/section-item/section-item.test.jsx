import { shallow } from 'enzyme';
import React from 'react';
import { SectionItem } from './section-item.component';

describe('SectionItem component', () => {
  let wrapper;
  let snapshot;
  let mockAddItem;
  const imageUrl = 'www.coolimage.com';
  const name = 'Boomer';

  beforeAll(() => {
    mockAddItem = jest.fn();
    const mockProps = {
      addItem: mockAddItem,
      item: {
        name: name,
        price: 25,
        imageUrl: imageUrl
      }
    };
    wrapper = shallow(<SectionItem { ...mockProps } />);
    snapshot = wrapper.debug();
  });

  it('should render SectionItem component',() => {
    expect( snapshot ).toMatchSnapshot();
  });

  it('should render the item with proper background image',() => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe( imageUrl );
  });

  it('should call addItem when clicked',() => {
    wrapper.find('AddButton').simulate('click');
    expect( mockAddItem ).toHaveBeenCalled();
  });

  it('should have proper name for NameContainer',() => {
    const testName = wrapper.find('NameContainer').text();
    expect( testName ).toBe( name );
  });
})
