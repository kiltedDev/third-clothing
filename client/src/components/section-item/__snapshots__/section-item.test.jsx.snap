// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionItem component should render SectionItem component 1`] = `
"<styled.div className=\\"section-item\\">
  <BackgroundImage className=\\"image\\" imageUrl=\\"www.coolimage.com\\" />
  <styled.div>
    <NameContainer>
      Boomer
    </NameContainer>
    <styled.span>
      25
    </styled.span>
  </styled.div>
  <AddButton onClick={[Function: onClick]} inverted={true}>
    Add to cart
  </AddButton>
</styled.div>"
`;
